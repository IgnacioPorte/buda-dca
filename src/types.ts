type HistoricalPrice = {
  price: number;
  date: string;
};

type MonthlyDCAInvestment = {
  price: number;
  date: string;
  fiatInvestment: number;
  cryptoPortfolio: number;
  portfolioFiatValue: number;
  gain: number;
  gainPercentage: number;
};


export type { HistoricalPrice, MonthlyDCAInvestment };