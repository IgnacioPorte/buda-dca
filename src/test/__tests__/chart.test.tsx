import LinearChart from "@/components/chart";
import { MonthlyDCAInvestment } from "@/types";
import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";

const mockData: MonthlyDCAInvestment[] = [
  {
    cryptoPortfolio: 0.1,
    date: "2021-01",
    fiatInvestment: 100,
    gain: 0,
    gainPercentage: 0,
    portfolioFiatValue: 150,
    price: 150,
  },
  {
    cryptoPortfolio: 0.2,
    date: "2021-02",
    fiatInvestment: 200,
    gain: 50,
    gainPercentage: 25,
    portfolioFiatValue: 250,
    price: 250,
  },
];

describe("LinearChart Component Tests", () => {
  test("renders without crashing", () => {
    expect(render(<LinearChart data={[]} />)).toMatchSnapshot();
  });

  test("renders chart with data", () => {
    expect(<LinearChart data={mockData} />).toMatchSnapshot();
  });
});
