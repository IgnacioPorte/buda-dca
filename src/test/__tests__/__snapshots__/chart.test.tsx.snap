// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinearChart Component Tests > renders chart with data 1`] = `
<LinearChart
  data={
    [
      {
        "cryptoPortfolio": 0.1,
        "date": "2021-01",
        "fiatInvestment": 100,
        "gain": 0,
        "gainPercentage": 0,
        "portfolioFiatValue": 150,
        "price": 150,
      },
      {
        "cryptoPortfolio": 0.2,
        "date": "2021-02",
        "fiatInvestment": 200,
        "gain": 50,
        "gainPercentage": 25,
        "portfolioFiatValue": 250,
        "price": 250,
      },
    ]
  }
/>
`;

exports[`LinearChart Component Tests > renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="md:w-2/3 w-full flex flex-col items-center"
      >
        <div
          class="animate-pulse bg-primary/10 h-full w-full rounded-xl"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="md:w-2/3 w-full flex flex-col items-center"
    >
      <div
        class="animate-pulse bg-primary/10 h-full w-full rounded-xl"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
